src/internal/connector/discovery/api/users.go

```diff
@@ -108,16 +108,19 @@ func (c Users) GetAll(ctx context.Context, errs *fault.Errors) ([]models.Userabl
 		return nil, clues.Wrap(err, "creating users iterator").WithClues(ctx).With(graph.ErrData(err)...)
 	}
 
-	us := make([]models.Userable, 0)
+	var (
+		us = make([]models.Userable, 0)
+		et = errs.Tracker()
```

Comment:
I know it's kind of sixes, but thoughts on passing in a tracker vs. making a new one in a function?

Reply:
It's not sixes, and would nullify the purpose of tracker.  If it were sixes, we'd be able to stick with the fault.Errors at large.  But that will cause the code to confuse global with local errors.  Good for failing out as fast as feasibly possible (honestly not a real concern, imo), then in trade we risk mass duplication of errors, which I think will do more harm overall.

By forcing new tracker generation in each func, we can ensure that only local errors are returned, thus protecting us against duplicate return values.

Reply:
ah, yea sorry, I didn't explain that clearly. I meant passing a tracker instance down 😅 

If it were a tracker instance passed in then maybe down the road we could do something like define an interface that both `fault.Errors` and the `fault.Tracker` implement and just pass that around. Though after saying that I'm not sure if that would make things more or less error prone... 🤔 idk I guess probably not the best idea now that I've given it more thought haha

for example:
```go
type bus interface {
  Err() error
  Add(err error)
}

func foo(errs bus) error {
  // do stuff
  if err != nil {
    errs.Add(err)
  }

  return errs.Err()
}
```

Reply:
> maybe down the road we could do something like define an interface that both fault.Errors and the fault.Tracker implement and just pass that around

I started things off with an interface like that.  Ultimately it turned out that the interface was almost exactly the fault.Errors interface, not a subset, and without a dependency to mock there wasn't any point to the abstraction.

It's important to note that a Tracker should be spawned for _each nested point of recoverable errors_.  Ie:
```go
func turtles(errs *fault.Errors) error {
  trk := errs.Tracker()
  if err := allTheWay(errs); err != nil {
    trk.Add(err)
  }
  return  trk.Err()
}

func allTheWay(errs *fault.Errors) error {
  trk := errs.Tracker()
  if err := down(errs); err != nil {
    trk.Add(err)
  }
  return  trk.Err()
}

func down(errs *fault.Errors) error {
  trk := errs.Tracker()
  if err := etc(errs); err != nil {
    trk.Add(err)
  }
  return  trk.Err()
}
```